### PR TITLE
Serverless Components + custom CloudFormation Resources

### PR DESCRIPTION
**Pull request by pmuens**
## What did you implement

This is an experiment in which we use Serverless Components within custom CloudFormation resources. Using such a combination will unlock a lot of interesting features in the future + we can do code-sharing (write once, use "everywhere").

## How can we verify it

There's a `README.md` file in this PR which describes how it can be tested and used. The usage pattern might change so I refer to this `.md` file rather than constantly updating this PR description with the most recent version of such `.md` file.

## Todos

<details>
<summary>Useful Scripts</summary>

- `npm run test-ci` --> Run all validation checks on proposed changes
- `npm run lint-updated` --> Lint all the updated files
- `npm run lint:fix` --> Automatically fix lint problems (if possible)
- `npm run prettier-check-updated` --> Check if updated files adhere to Prettier config
- `npm run prettify-updated` --> Prettify all the updated files

</details>

- [ ] Write and run all tests
- [ ] Write documentation
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

**_Is this ready for review?:_** NO
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: cf-via-components/code/handler.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+const { Context, extractInputs, extractState, handlerWrapper } = require('./utils');
+
+function handler(event) {
+  const inputs = extractInputs(event);
+  // NOTE: apparently OldResourceProperties is only passed-in when
+  // dealing with Update requests --> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requests.html
+  // therefore we re-use the inputs as 'state' when dealing with `Delete` requests
+  let state = {};
+  if (event.RequestType === 'Update') {
+    state = extractState(event);
```
Comment (medikoo) [inline cf-via-components/code/handler.js]: In my understanding what is extracted here is not really a _state_, but previous _input_ (one that was provided as _input_ to custom resource with previous successful stack deployment), so it's not a value in any way output by component.

When we're after a _state_, I think we need to, store, output from component, somewhere in S3 (or DynamoDb) storage,  and with every consecutive deployment retrieve it, pass it to component, and update it with value returned by it (at least that's how I think generic approach to components state handling could look).

But, maybe I misunderstood something?

**Comment by pmuens** [comment]
Thanks for the review @medikoo :+1:

Yes, this is just a hack and I committed and pushed it so that it's backed-up on GitHub and does not solely reside on my local machine. I'll refactor this once the core value proposition is validated.

**Comment by pmuens** [comment]
Here's just a quick update that we're currently investigating how we want to integrate Serverless Components into Serverless Framework v1.

Given that, this PR is on hold for the time-being...

**Comment by pmuens** [comment]
Closing this for now as it was just a "throwaway PR" which we've used to work on a PoC on how to re-use Serverless Components in Serverless Framework v1.

We might re-open this PR once we continue our efforts to introduce existing components into v1.

**Pull request closed without merging**